onnxruntime/core/framework/tensor_type_and_shape.h

```diff
@@ -2,25 +2,50 @@
 // Licensed under the MIT License.
 #pragma once
 
+#include <memory>
 #include <string>
 #include <vector>
 
 #include "core/framework/tensor_shape.h"
 #include "core/session/onnxruntime_c_api.h"
 
+namespace ONNX_NAMESPACE {
+class TypeProto;
+}
+
+namespace onnxruntime {
+class DataTypeImpl;
+}
+
 struct OrtTensorTypeAndShapeInfo {
  public:
+  using Ptr = std::unique_ptr<OrtTensorTypeAndShapeInfo>;
```

Comment:
nit: IMHO this alias unnecessarily abstracts what sort of pointer is involved, which makes it harder to understand code that uses it, as it's not clear if it's a raw pointer, unique pointer or shared pointer. someone reviewing the usage code also has to go and find this `using` statement to discover it's a unique_ptr and no call to 'delete' was required.

IIRC you updated a bunch of places where IAllocatorPtr was unnecessarily passed by value, because the fact it was a shared_ptr was hidden by the alias and the cost of the reference count increment wasn't clear. This alias creates that same sort of issue.

Reply:
I do not recall any shared_ptr in the past. 
I felt this change would make the code more robust.
It did need to allocate using smart pointers and it did not need to return 'no so smart' OrtStatus, because this code itself is not a public API.

Reply:
Ah, now I remember.